### PR TITLE
Some dna fixes

### LINKED ISSUE #7597: Genetic aide is manifesting on monkeys.
As stated, the sprites don't match and leads to some interesting views of monkeys with giant glows around them and green people with tails who quickly turn back into monkeys.

Possibly intended.....?

### LINKED ISSUE #9083: Genetics disabling viruses
Being infected with viruses and then turned into monkey and back to human disables the viruses.
The viruses still exist and say they're 5/5 stages but they do absolutely nothing.

### LINKED ISSUE #10355: Space cleaner can make genetic powers invisble.
I was working on powers in genetics when a drone came in and sprayed me with some cleaner to clean my bloody lab coat.This made all the powers I currently had (hulk and cold resistance) invisible. Possible you could use the shower to same effect as well.

### PR DESCRIPTION
**Pull request by phil235**
Fixes monkeyize/humanize removing the mob's viruses when they should be kept.
Fixes NOCLONE and CLUMSY being in both disabilities and mutations. they're now only disabilities.
Fixes minttoxin not gibbing people with the fat disabilities.
Fixes some runtimes, replacing usr by user in some places.
Fixes mutation overlays being removed when updating overlays.
Remove the now unused mob/var/list/mutations and human/var/blood_type
Fixed some formatting in preferences.dm

Fixes #7597
Fixes #9083
Fixes #10355

### CODE REVIEW

File: code/modules/client/preferences.dm [hunk 1]
```diff
@@ -117,984 +117,983 @@ var/global/list/special_roles = list( //keep synced with the defines BE_* in set
...
-
-						var/result = input(user, "Select a species", "Species Selection") as null|anything in roundstart_species
-
-						if(result)
-							var/newtype = roundstart_species[result]
-							pref_species = new newtype()
-							//Now that we changed our species, we must verify that the mutant colour is still allowed.
-							var/temp_hsv = RGBtoHSV(features["mcolor"])
-							if(features["mcolor"] == "#000" || (!(MUTCOLORS_PARTSONLY in pref_species.specflags) && ReadHSV(temp_hsv)[3] < ReadHSV("#7F7F7F")[3]))
-								features["mcolor"] = pref_specie
```
Comment (phil235) [inline code/modules/client/preferences.dm]: The only real change done in this file. I removed this line. blood_type is now only in dna and updated here in the hardset_dna() on line 1096.

**Comment by WJohn** [comment]
Oh my god.

**Comment by Razharas** [comment]
>Fixes mutation overlays being removed when updating overlays.

YES OH GOD YES ITS DONE IM FREE NOW WOOOOO

File: code/datums/mutations.dm [hunk 2]
```diff
@@ -147,6 +142,19 @@
...
+
+/datum/mutation/human/telekinesis/on_acquiring(mob/living/carbon/human/owner)
+	if(..())
+		return
+	owner.update_mutation_overlays()
+
+/datum/mutation/human/telekinesis/on_losing(mob/living/carbon/human/owner)
+	if(..())
+		return
+	owner.update_mutation_overlays()
```
Comment (Razharas) [inline code/datums/mutations.dm]: No need to do it per-mutation basis, add to parent

File: code/datums/mutations.dm [hunk 3]
```diff
@@ -49,21 +49,14 @@
 		. = on_losing(owner)
 
 /datum/mutation/human/proc/on_acquiring(mob/living/carbon/human/owner)
-	if(!owner || (src in owner.dna.mutations))
+	if(!owner || !istype(owner) || (src in owner.dna.mutations))
```
Comment (phil235) [inline code/datums/mutations.dm]: so monkeys don't acquire mutations. (exception made for race mut currently)

File: code/datums/mutations.dm (hunk 2 above)
Comment (phil235) [inline code/datums/mutations.dm]: will change that, yeah

File: code/datums/spells/genetic.dm [hunk 4]
```diff
@@ -18,6 +18,8 @@
 /obj/effect/proc_holder/spell/targeted/genetic/cast(list/targets)
 	playMagSound()
 	for(var/mob/living/carbon/target in targets)
+		if(!target.dna)
+			continue
```
Comment (phil235) [inline code/datums/spells/genetic.dm]: fixes potential runtime, not all carbons have non null dna.

File: code/modules/admin/create_mob.dm [hunk 5]
```diff
@@ -21,5 +20,6 @@
 	H.facial_hair_color = H.hair_color
 	H.eye_color = random_eye_color()
 	H.dna.blood_type = random_blood_type()
+	H.dna.generate_uni_identity(H)
```
Comment (phil235) [inline code/modules/admin/create_mob.dm]: Moved the UI generation at the end so the UI matches the new body characteristics.

File: code/modules/mob/living/carbon/human/human.dm [hunk 6]
```diff
@@ -29,8 +29,9 @@
 		I.Insert(src)
 
 	// for spawned humans; overwritten by other code
-	ready_dna(src)
+	create_dna(src)
 	randomize_human(src)
+	ready_dna(src)
```
Comment (phil235) [inline code/modules/mob/living/carbon/human/human.dm]: so admin spawned human's UI matches their body.

File: code/modules/mob/living/carbon/human/human_defines.dm [hunk 7]
```diff
@@ -18,7 +18,6 @@
 	var/lip_color = "white"
 
 	var/age = 30		//Player's age (pure fluff)
-	var/blood_type = "A+"	//Player's bloodtype
```
Comment (phil235) [inline code/modules/mob/living/carbon/human/human_defines.dm]: so we don't have two blood_type vars (one in human, on in the dna)

File: code/modules/mob/living/carbon/human/human_helpers.dm [hunk 8]
```diff
@@ -56,7 +56,7 @@
 	if( head && (head.flags_inv&HIDEFACE) )
 		return if_no_face		//Likewise for hats
 	var/obj/item/organ/limb/O = get_organ("head")
-	if( (status_flags&DISFIGURED) || (O.brutestate+O.burnstate)>2 || cloneloss>50 || !real_name )	//disfigured. use id-name if possible
+	if( !O || (status_flags&DISFIGURED) || (O.brutestate+O.burnstate)>2 || cloneloss>50 || !real_name )	//disfigured. use id-name if possible
```
Comment (phil235) [inline code/modules/mob/living/carbon/human/human_helpers.dm]: prevent potential runtime, get_organ("head") can return null.

File: code/modules/mob/mob_defines.dm [hunk 9]
```diff
@@ -106,9 +106,6 @@
 
 	var/radiation = 0//Carbon
 
-	var/list/mutations = list() //Carbon -- Doohl
-	//see: setup.dm for list of mutations
-
```
Comment (phil235) [inline code/modules/mob/mob_defines.dm]: mutations are now listed in the dna, not the mob.

File: code/modules/mob/mob_transformation_simple.dm [hunk 10]
```diff
@@ -40,7 +40,7 @@
 		M.name = src.name
 		M.real_name = src.real_name
 
-	if(check_dna_integrity(src) && istype(M, /mob/living/carbon))
+	if(check_dna_integrity(src))
```
Comment (phil235) [inline code/modules/mob/mob_transformation_simple.dm]: check_dna_integrity() already checks for carbon subtypes.

File: code/modules/mob/transform_procs.dm [hunk 11]
```diff
@@ -59,6 +59,7 @@
 		viruses = list()
 		for(var/datum/disease/D in O.viruses)
 			D.affected_mob = O
+			D.holder = O
```
Comment (phil235) [inline code/modules/mob/transform_procs.dm]: Without a holder, viruses can't do anything.

File: code/modules/mob/transform_procs.dm [hunk 12]
```diff
@@ -169,7 +170,6 @@
 	else
 		hardset_dna(O, null, null, null, null, dna.species.type)
 
-	dna = null
```
Comment (phil235) [inline code/modules/mob/transform_procs.dm]: Why was this ever here....

File: code/modules/ninja/ninja_event.dm [hunk 13]
```diff
@@ -167,7 +167,6 @@ Contents:
 
 /proc/create_space_ninja(spawn_loc)
 	var/mob/living/carbon/human/new_ninja = new(spawn_loc)
-	if(prob(50)) new_ninja.gender = "female"
```
Comment (phil235) [inline code/modules/ninja/ninja_event.dm]: gender already picked via the new preferences and the copy_to() below.

File: code/modules/reagents/Chemistry-Reagents/Toxin-Reagents.dm [hunk 14]
```diff
@@ -123,7 +123,7 @@
 	toxpwr = 0
 
 /datum/reagent/toxin/minttoxin/on_mob_life(mob/living/M)
-	if (FAT in M.mutations)
+	if (M.disabilities & FAT)
```
Comment (phil235) [inline code/modules/reagents/Chemistry-Reagents/Toxin-Reagents.dm]: FAT is a disabilities, not a mutation.

**Comment by PKPenguin321** [comment]
>Fixes minttoxin not gibbing people with the fat disabilities.

FINALLY
MY REIGN OF MINTTOXIN IN THE CRYO TUBES AND A WEIGHT GAIN VIRUS CAN TRULY BEGIN

**Comment by PKPenguin321** [comment]
also, #11768 also claims to fix #10355, so i'm guessing it'll conflict with this

**Comment by Razharas** [comment]
Is it ready?

**Comment by phil235** [comment]
@Razharas Yes, ready to merge.

**Pull request merged**